flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/ExecutionGraphRestartTest.java

```diff
@@ -661,10 +682,117 @@ public void testConcurrentGlobalFailAndRestarts() throws Exception {
 		}
 	}
 
+	@Test
+	public void testRestartWithEagerSchedulingAndSlotSharing() throws Exception {
```

Comment:
Does this test just further verify existing behaviour? I'm asking because it doesn't fail when changing the order of slot release and restart back to the original order.